src/issue-strategies/bug-maintenance.js

```diff
@@ -1,7 +1,18 @@
+import * as jiraOperations from '../jira-operations.js';
+
 export function apply(issue, jiraClientAPI) {
   if(issue === null || issue.fields.status.statusCategory.colorName !== 'yellow') {
     return Promise.reject(new Error(`Issue ${issue.key} is not open to commit against`));
   }
 
+  if(issue.fields.issuelinks){
+    return jiraOperations.findParent(issue, jiraClientAPI).then((parent)=>{
+      if(!parent){
+        return Promise.resolve(true);
+      }
+      return parent.fields.status.statusCategory.colorName === 'yellow' ? Promise.resolve(true) : Promise.reject(new Error(`Cannot commit issue ${issue.key} because parent issue ${parent.key} is not available to commit against.`));
+    });
+  }
+
   return Promise.resolve(true);
```

Comment:
To add to what @aabenoja said, you generally want to avoid ternary operations that are long and complicated. They make it difficult to read the code. Generally you only want to use ternaries when it's really simple. For example:

```
// Bad
return parent.fields.status.statusCategory.colorName === 'yellow' ? Promise.resolve(true) : Promise.reject(new Error(`Cannot commit issue ${issue.key} because parent issue ${parent.key} is not available to commit against.`));

// Good
var color = x.success ? 'green' : 'red'
```

Note how that good example can be easily read within a second, whereas with your example there's so many characters in the way that it's hard to quickly see where the ternary separaters are.